psetv2/input.go

```diff
@@ -690,6 +712,66 @@ func (i *Input) getKeyPairs() ([]KeyPair, error) {
 		keyPairs = append(keyPairs, issuanceBlindInflationKeysProofKeyPair)
 	}
 
+	if i.ExplicitValue != 0 {
+		var explicitValueBytes []byte
+		binary.LittleEndian.PutUint64(explicitValueBytes, i.ExplicitValue)
+
+		explicitValueKeyPair := KeyPair{
+			Key: Key{
+				KeyType: PsetProprietary,
+				KeyData: proprietaryKey(InputExplicitValue, nil),
+			},
+			Value: explicitValueBytes,
+		}
+		keyPairs = append(keyPairs, explicitValueKeyPair)
+	}
+
+	if len(i.ValueProof) > 0 {
+		valueProofKeyPair := KeyPair{
+			Key: Key{
+				KeyType: PsetProprietary,
+				KeyData: proprietaryKey(InputValueProof, nil),
+			},
+			Value: i.ValueProof,
+		}
+		keyPairs = append(keyPairs, valueProofKeyPair)
+	}
+
+	if len(i.ExplicitAsset) > 0 {
+		explicitAssetKeyPair := KeyPair{
+			Key: Key{
+				KeyType: PsetProprietary,
+				KeyData: proprietaryKey(InputExplicitAsset, nil),
+			},
+			Value: i.ExplicitAsset,
+		}
+		keyPairs = append(keyPairs, explicitAssetKeyPair)
+	}
+
+	if len(i.AssetProof) > 0 {
+		assetProofKeyPair := KeyPair{
+			Key: Key{
+				KeyType: PsetProprietary,
+				KeyData: proprietaryKey(InputAssetProof, nil),
+			},
+			Value: i.AssetProof,
+		}
+		keyPairs = append(keyPairs, assetProofKeyPair)
+	}
+
+	blindedIssuance := []byte{0}
+	if i.BlindedIssuance {
+		blindedIssuance = []byte{1}
+	}
+	assetProofKeyPair := KeyPair{
+		Key: Key{
+			KeyType: PsetProprietary,
+			KeyData: proprietaryKey(InputBlindedIssuanceValue, nil),
+		},
+		Value: blindedIssuance,
+	}
```

Comment:
The spec says:
> A boolean flag. 0x00 indicates the issuance should not be blinded, 0x01 indicates it should be. If not specified, assumed to be 0x01. Note that this does not indicate actual blinding status, but rather the expected blinding status prior to signing.

To correctly follow the spec we have to declare `BlindedIssuance` as `*bool`. This way, in case it's not defined we can default to 0x01.

Reply:
Also, with this kind of declaration we shouldn't break unit tests because in case the field is nil we just don't serialize it.

Reply:
Done